### PR TITLE
#1078 - Create custom report

### PR DESCRIPTION
**Pull request by HalcyonJAC**
## What's included?
A new way to inspect data within the dashboard and generate / save your own reports.

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?
Go to an Exercise
Click the reports tab
Click the Custom report link on the left nav

Functionality:
1. Select a column from the dropdown to show the values in a table
2. Add multiple columns, delete them by clicking and re-order by dragging.
3. Change the report type to count to view aggregated results
4. On the count report, click one of the values to add a 'where clause' which limits the results
5. Save a report by clicking Save Report at the top right of the page, give it a name. Then switch to a different exercise and try to load the report.

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

### CODE REVIEW

File: src/views/Exercise/Reports/Custom.vue [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+import Modal from '@jac-uk/jac-kit/components/Modal/Modal';
+import LoadingMessage from '@jac-uk/jac-kit/draftComponents/LoadingMessage';
+
+export default {
+  components: {
+    Modal,
+    draggable,
+    LoadingMessage,
+  },
+  data() {
```
Comment (tomlovesgithub) [inline src/views/Exercise/Reports/Custom.vue:288]: `isLoading` and `loadFailed` need to be added to the data here.

**Review by tomlovesgithub — CHANGES_REQUESTED** [review]
still getting console errors when testing locally ;?

**Review by warrensearle — APPROVED** [review]
Awesome 👍

**Review by mbrookeswebdev — APPROVED** [review]

File: src/views/Exercise/Reports/Custom.vue (hunk 1 above)
Comment (HalcyonJAC) [reply src/views/Exercise/Reports/Custom.vue:288]: Hi,

This data has been added to the component.

**Review by joy-ade — COMMENTED** [review]
feedbacks from UTG  "select a column to display" - can that be re-ordered to have application info 1st maybe as its likely the most people would filter on his on the application status

**Comment by joy-ade** [comment]
@HalcyonJAC @JeremykJAC Can you include and export button on this report pls.

**Review by joy-ade — COMMENTED** [review]
Feedback form UTG Can we download this report pls.

**Comment by HalcyonJAC** [comment]
@joy-ade Hi - I've made some of the changes but I think there might be some more work to do on this later depending.

- Order of options is changed
- Added jurisdiction and location
- fixed a bug with saving reports
- added some validation on saving reports
- adding a message to show the report was saved

**Review by joy-ade — APPROVED** [review]
Approved by Product team and UTG

**Review by tomlovesgithub — APPROVED** [review]

**Pull request merged**